pkgs/dart_mcp/lib/src/api.dart

```diff
@@ -0,0 +1,1300 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
```

Comment:
Note that there are many commented out typescript definitions here - these are the ones I haven't yet ported.

We likely want a code generator, I ported just enough to get the steel thread sort of through here, but its easiest to understand the current state to just leave in these un-ported types as comments.